platform/machine/packet/machine.go

```diff
@@ -65,8 +65,17 @@ func (pm *machine) Reboot() error {
 }
 
 func (pm *machine) Destroy() {
-	if err := pm.cluster.flight.api.DeleteDevice(pm.ID()); err != nil {
-		plog.Errorf("Error terminating device %v: %v", pm.ID(), err)
+	// Instead of actually deleting the device.
+	// We add it to the devices pool in order to mark it
+	// as "ready to be used" by other tests.
+	id := pm.ID()
+	pm.cluster.flight.devicesPool <- id
+	plog.Infof("device %s added to the pool", id)
+
+	// console needs to be manually closed in order to prevent program from
+	// freezing on `done` channel in the `Output()` console's method.
+	if err := pm.console.Close(); err != nil {
```

Comment:
I think we should close the session here instead of the console. The `startConsole` function waits on `session.Wait()` and will take care of closing the file writer and the console.

Reply:
I realize that this needs a bit of rework in `startConsole` (`api/packet/api.go`) to expose a way of closing the session

Reply:
So I rewrote a bit in order to keep a map of opened SSH clients and we close the associated client when we mark the device as available.